src/coreclr/tools/Common/TypeSystem/Common/MetadataFieldLayoutAlgorithm.cs

```diff
@@ -349,6 +352,7 @@ protected static ComputedInstanceFieldLayout ComputeExplicitFieldLayout(Metadata
                     );
                 if (needsToBeAligned)
                 {
+                    largestAlignmentRequired = LayoutInt.Max(largestAlignmentRequired, type.Context.Target.LayoutPointerSize);
```

Comment:
I would expect that `ComputeFieldSizeAndAlignment` will return sufficient alignment for GC pointers or anything with GC pointers. Is this really needed? If yes, it would be worth a comment explaining the situation where it is needed.

Reply:
I believe you're right, I have replaced this with just an assert double-checking that at this point <code>largestAlignmentRequired</code> must be greater than or equal to <code>PointerSize</code>.